### PR TITLE
feat(connector): [Multisafepay] Use transaction_id as reference to transaction

### LINKED ISSUE #2336: [FEATURE]: [Multisafepay] Use `connector_response_reference_id` as reference to merchant
### :memo: Feature Description

- Reference id are used to map transactions in the connector’s dashboard.
- Hyperswitch manages several reference ids, such as `payment_id`, `attempt_id`, and `connector_transaction_id` for a single transaction. 
- However, merchants may encounter uncertainty when determining which ID to utilize in the connector dashboard to identify the payment.

### :hammer: Possible Implementation

- When we receive a response from the connector for the payment, we deserialize it and populate the `response` field in RouterData. 
- For the `TransactionResponse` type, we must fill the `connector_response_reference_id` with a corresponding reference id for the merchant to identify the transaction. 
- One might need to have exposure to api docs of the connector for which it is being implemented to decide what to fill in connector_response_reference_id.
- You can check this PR for further reference https://github.com/juspay/hyperswitch/pull/1735

🔖 Note: All the changes needed should be contained within `hyperswitch/crates/router/src/connector/`

### :package: Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### :package: Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by nabeel001**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Updated the `connector_response_reference_id` value from the response.
closes #2336 

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
- Reference id are used to map transactions in the connector’s dashboard.
- Hyperswitch manages several reference ids, such as payment_id, attempt_id, and connector_transaction_id for a single transaction.
- However, merchants may encounter uncertainty when determining which ID to utilize in the connector dashboard to identify the payment.

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/multisafepay/transformers.rs [hunk 1]
```diff
@@ -543,7 +543,7 @@ impl<F, T>
                     }),
                 connector_metadata: None,
                 network_txn_id: None,
-                connector_response_reference_id: None,
+                connector_response_reference_id: item.response.data.transaction_id,
```
Comment (srujanchikke) [inline crates/router/src/connector/multisafepay/transformers.rs:546]: Hello @nabeel001 ,

As per api docs of [multisafepay](https://docs.multisafepay.com/reference/createorder) , `order_id` is merchant_refernece_id which should be mapped to `connector_reference_id`

Comment (nabeel001) [reply crates/router/src/connector/multisafepay/transformers.rs:546]: @srujanchikke 
I had misunderstood the issue's requirement and mapped the `transaction_id` directly from the response.
Thanks for pointing it out, I'll make the required change.

image

**Comment by srujanchikke** [comment]
Hello @nabeel001 

Please use following cargo commands -

`cargo check --all-features`                                                                 
`cargo clippy --all-features`
ref -[ Cargo commands](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md#cargo-commands), and fix the errors thrown by clippy for the checks to pass .

**Comment by srujanchikke** [comment]
Hey @nabeel001 ,

Please update the branch !

**Comment by VedantKhairnar** [comment]
Hey @nabeel001 ,
Let us know if you face any issues. Happy to help! : )

**Comment by nabeel001** [comment]
@srujanchikke @VedantKhairnar I have updated the code and all the tests are passing too.
kindly review and merge.

**Review by srujanchikke — APPROVED** [review]
Looks good to me!

**Review by SanchithHegde — APPROVED** [review]
Looks good to me!

Thanks for the PR, @nabeel001!

**Comment by VedantKhairnar** [comment]
Hey @nabeel001  👋,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.
Thanks!

**Comment by nabeel001** [comment]
@VedantKhairnar 
I have filled the google form using another mail Id that is not associated with my GitHub account. Is that fine or should I submit another response with the mail id that is linked to my Github account?
can u pls confirm this..

**Comment by VedantKhairnar** [comment]
That's fine @nabeel001 .
You don't need to refill.
.
Thanks for your contribution! : )

**Comment by nabeel001** [comment]
@VedantKhairnar Any Idea when will I be receiving my swag, I'm asking because I haven't received any mail about it either.

**Comment by VedantKhairnar** [comment]
Hey @nabeel001 
Printing and shipping is in process. Some of them are shipped. It might take around a week more. 
Thanks for your patience. Let us know if you have any other queries.

**Pull request merged**